Diplom-2 task: Created API-tests with AllureReport

SergeyDyomin92 opened the pull request:
Написаны API-автотесты для проекта StellarBergers. Покрыты тестами методы:
- POST /api/auth/register
- POST /api/auth/login
- PATCH /api/auth/user
- POST api/orders
- GET api/orders.

После каждого теста, сгенерированный пользователь удалялся по DELETE запросу /api/auth/user.
Создан Allure-отчёт по всем проверкам. В результате работы удалось определить расхождения ожидаемого поведения api-документации и реализации. По найденным ошибкам можно завести баги в багтрекинговой системе.

Используемые библиотеки и плагины: 
- junit 4.13.1
- aspectj 1.9.7
- allure 2.15.0
- rest-assured 5.4.0
- java-faker 1.0.2

SergeyDyomin92 committed:
Diplom-2 task: Created API-tests with AllureReport

nikolay-977 commented on src/test/java/CreateUserTests.java:
```diff
@@ -0,0 +1,133 @@
+    }
+
+    @Step("Отправить POST запрос /api/auth/register")
+    public Response sendPostAPIAuthRegister(User user) {
```
Необходимо исправить: В src.main.java.api.client для каждой ручки создать классы UserClient(для /api/auth) и OrdersClient(для /api/orders). Вынести реализацию шагов в эти классы

Можно улучшить: применить наследование - создать класс BaseClient, в нем реализовать метод  с RequestSpecification. Остальные клиентские классы наследовать от BaseClient. Указывать BASE_URL  и вызывать лог лучше в RequestSpecification, это позволит избежать повторяющегося кода.

nikolay-977 commented on src/test/java/GetUserOrdersTests.java:
```diff
@@ -0,0 +1,95 @@
+import static models.UserGenerator.randomUser;
+import static org.hamcrest.Matchers.equalTo;
+
+public class GetUserOrdersTests {
```
Можно улучшить: применить наследование - создать класс BaseTest и вынести в него методы с аннотациями `@Before` и `@After`, где это возможно. Это позволит избежать повторяющегося кода и сократит код автотестов.

nikolay-977 reviewed (commented):
Можно улучшить: хорошей практикой является добавление README.md с описанием проекта, которое включает название, технологии в проекте (например, Java 11, JUnit 4.13.2, maven 3.9.0 и т.д.), настройка (если необходима) и как запускать (`mvn clean test` - кавычки позволяют выполнять запуск из файла README.md). Для оформления данных в виде таблице используй сочетание клавиш Ctrl + N или Cmd + N. Как оформлять README.md можно ознакомиться в статье https://blog.jetbrains.com/idea/2022/08/markdown-support-improvements/

SergeyDyomin92 committed:
Diplom-2 task: Исправлено замечание: В src.main.java.api.client для каждой ручки созданы классы UserClient(для /api/auth) и OrdersClient(для /api/orders). Вынесена реализация шагов в эти классы.
 Добалвено улучшение: создан класс BaseClient, в нем реализован метод с RequestSpecification. Остальные клиентские классы унаследованы от BaseClient.
 Добалвено улучшение: заполнен файл README.md. Тесты перезапущены и сгенерирован новый отчет.
